### ISSUE TITLE
[FR] first class module support

### ISSUE THREAD

**Issue by FossPrime** [feathersjs/feathers-chat#220]
Reproduction: [stackblitz sandbox](https://stackblitz.com/edit/feathers-chat?file=src%2Fservices%2Fmessages%2Fmessages.schema.ts%3AL5,tsconfig.json,src%2Fservices%2Fusers%2Fusers.service.ts%3AL5,package.json%3AL18,tsconfig.node.json%3AL5)
git repo: https://github.com/feathersjs/playground/tree/main/vite-chat

Modules are well supported everywhere that matters since TypeScript 4.7. There are no hard hurdles, just lots of tedious stuff that would get missed if we don't use it/test it.

basically the code that works in module only mode, works in CommonJs Typescript mode, but not vise versa. Practically speaking... its a matter of using subpath imports, '.js' as the extension, and pointing directories to their index.ts file (instead of having the loader check if there is a file/directory and then resolve to index.ts if there is a directory.).

**Comment by FossPrime** [comment]
This is the wrong place for this. This involves large scale refuckulation of feathersjs/feathers.
